### ISSUE TITLE
[stable/airflow] extra new line in database connection strings

### ISSUE THREAD

**Issue by jameswynn** [helm/charts#23942]
**Describe the bug**
When using external database connection, the following variables have additional newline characters appended:
* DATABASE_PASSWORD_CMD
* DATABASE_SQLALCHEMY_CMD
* DATABASE_CELERY_CMD

Relevant stacktrace:
```
[2020-10-22 12:27:09,036] {taskinstance.py:1150} ERROR - Cannot execute bash -c 'eval "echo -n "postgresql+psycopg2://airflow@dbserver:$(eval echo ${DATABASE_PASSWORD} | python3 -c "import urllib.parse; encoded_pass = urllib.parse.quote(input()); print(encoded_pass)"
)@x.x.x.x:5432/airflow?sslmode=require
"
"'
. Error code is: 127. Output: postgresql+psycopg2://airflow@dbserver:@x.x.x.x:5432/airflow?sslmode=require, Stderr: bash: line 3: $'\n': command not found
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/subdag_operator.py", line 102, in execute
    executor=self.executor)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dag.py", line 1432, in run
    job.run()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/jobs/base_job.py", line 218, in run
    self._execute()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/db.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/jobs/backfill_job.py", line 787, in _execute
    session=session)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/db.py", line 70, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/jobs/backfill_job.py", line 717, in _execute_for_run_dates
    session=session)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/db.py", line 70, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/jobs/backfill_job.py", line 586, in _process_backfill_task_instances
    _per_task_process(task, key, ti)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/db.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/jobs/backfill_job.py", line 497, in _per_task_process
    cfg_path = tmp_configuration_copy()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/configuration.py", line 35, in tmp_configuration_copy
    cfg_dict = conf.as_dict(display_sensitive=True, raw=True)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/configuration.py", line 574, in as_dict
    opt = self._get_cmd_option(section, key)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/configuration.py", line 304, in _get_cmd_option
    return run_command(command)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/configuration.py", line 99, in run_command
    .format(command, process.returncode, output, stderr)
airflow.exceptions.AirflowConfigException: Cannot execute bash -c 'eval "echo -n "postgresql+psycopg2://airflow@dbserver:$(eval echo ${DATABASE_PASSWORD} | python3 -c "import urllib.parse; encoded_pass = urllib.parse.quote(input()); print(encoded_pass)"
)@10.98.32.5:5432/airflow?sslmode=require
"
"'
. Error code is: 127. Output: postgresql+psycopg2://airflow@dbserver:@x.x.x.x:5432/airflow?sslmode=require, Stderr: bash: line 3: $'\n': command not found
```

**Version of Helm and Kubernetes**:
Helm 3.2.4
Kubernetes 1.17.5

**Which chart**:
stable/airflow 7.11.0

**What happened**:
Configured the chart for external database connections. Most of the system works, but when executing DAGs it fails when constructing database connections stating there is a spurious newline character.

**What you expected to happen**:
External database is correctly connected.

**How to reproduce it** (as minimally and precisely as possible):
Configure an external database and run a DAG.

**Anything else we need to know**:
This appears to be because configmap-env.yaml is using "|" instead of "|-" to truncate the trailing newline.

**Comment by thesuperzapper** [comment]
I have raised a PR #23951 for this, but there must be something in my environment which is causing this not to be an issue.

**Comment by thesuperzapper** [comment]
This is now fixed in versions after `7.13.1`
